### ISSUE TITLE
Option to disable auto cache refresh in proxy for improved security 

### ISSUE THREAD

**Issue by sowhatim** [goharbor/harbor#16199]
**Is your feature request related to a problem? Please describe.**
We are using proxy cache feature and what If somebody plans uses an image's specific tag and does manual verification on his/her end. Now what if a vulnerable version gets pushed at the origin repository later and due to harbor proxy default refresh behaviour , our subsequent deployments are now having vulnerable version. This use-case where we also need to manually verify the image couldn't be solved via automated scanning. 

**Describe the solution you'd like**
We want option to completely disable refresh of any image version once pulled. 

**Describe the main design/architecture of your solution**
This will be like pull once and forget. If we have manually tested and verified the authenticity of an image version, we don't want to worry about it anytime again. 

**Describe the development plan you've considered**
It can be a feature of registry endpoint.  We set a registry endpoint and disable the auto-refresh feature of cache in it. If one wants the different behaviour for some other use-case, a separate registry endpoint and thus a separate project can be created for it with the feature being enabled. 

**Additional context**
This will be a very important addition to ensure that our container systems remains secure.

**Comment by zyyw** [comment]
Hi @sowhatim , the idea of `completed disable refresh of any image version once pulled` may not sound valid, in the case that the image to be refreshed doesn't contains any vulnerabilities at all, and it does expected to be refreshed so that the the latest image released are available in deployment.

Is it possible to improve your release process? for example, you may currently have a release, say v1.0.0, and after some development work, the images for the new release can be upgraded to v1.1.0 for example.
In your case, I assume it is image_name:latest that is used for deployment. And if vulnerabilities exist in a new release, you will have security concerns. But harbor honestly doesn't know whether it contains vulnerabilities or not BEFORE refresh the latest image and scan it.

**Comment by sowhatim** [comment]
@zyyw - yes, we do use the same release process and so I am talking about the same use-case. How do one ensure that if an image with version v1.0.0 is downloaded and verified manually, it won't change in the system. As when harbor refreshes it cache, I might have some different image running in the system for new Pods, right ?

**Comment by stonezdj** [comment]
It seems that tag immutability in the original registry server could solve the current problem, not the proxy cache. If the tag is immutable in the original registry, then the proxy cache won't pull the updated tag.

**Comment by sowhatim** [comment]
@stonezdj , you are right but one can't ensure the same for third party registry servers. And to keep your side of things safe, is there any other way in harbor where I can ensure that the image under the old tag is not changed ?

**Comment by mnacharov** [comment]
I have the same view on this. Harbor currently has "Tag Immutability" for proxy cache repos, but it doesn't work.
I would like to have this feature since this way, I can enable refresh for `:latest` and keep everything else immutable. This way, I'll be sure that the community repository owner won't be able to update any workload in my cluster without me knowing about it.
